Common/cpp/LayoutAnimations/LayoutAnimationsProxy.cpp

```diff
@@ -11,35 +11,117 @@ namespace reanimated {
   sv->removeListener(tag + idOffset);
-  observedValues.erase(tag);
-  this->notifyAboutEnd(tag, !finished);
+  observedValues_.erase(tag);
+  this->endHandler_(tag, !finished, removeView);
+}
+
+void LayoutAnimationsProxy::configureAnimation(
+    int tag,
+    const std::string &type,
+    std::shared_ptr<ShareableValue> config,
+    std::shared_ptr<ShareableValue> viewSharedValue) {
+  auto lock = std::unique_lock<std::mutex>(animationsMutex_);
+  if (type == "entering") {
+    enteringAnimations_[tag] = config;
+  } else if (type == "exiting") {
+    exitingAnimations_[tag] = config;
+  } else if (type == "layout") {
+    layoutAnimations_[tag] = config;
+  }
+  viewSharedValues_[tag] = viewSharedValue;
+}
+
+bool LayoutAnimationsProxy::hasLayoutAnimation(
+    int tag,
+    const std::string &type) {
+  auto lock = std::unique_lock<std::mutex>(animationsMutex_);
+  if (type == "entering") {
+    return enteringAnimations_.find(tag) != enteringAnimations_.end();
+  } else if (type == "exiting") {
+    return exitingAnimations_.find(tag) != exitingAnimations_.end();
+  } else if (type == "layout") {
+    return layoutAnimations_.find(tag) != layoutAnimations_.end();
+  }
+  return false;
+}
+
+void LayoutAnimationsProxy::clearLayoutAnimationConfig(int tag) {
+  auto lock = std::unique_lock<std::mutex>(animationsMutex_);
+  enteringAnimations_.erase(tag);
+  exitingAnimations_.erase(tag);
+  layoutAnimations_.erase(tag);
+  viewSharedValues_.erase(tag);
 }
 
-void LayoutAnimationsProxy::notifyAboutCancellation(int tag) {
-  this->notifyAboutEnd(tag, false);
+void LayoutAnimationsProxy::startLayoutAnimation(
+    jsi::Runtime &rt,
+    int tag,
+    const std::string &type,
+    const jsi::Object &values) {
+  std::shared_ptr<ShareableValue> config;
+  std::shared_ptr<ShareableValue> viewSharedValue;
+  {
+    auto lock = std::unique_lock<std::mutex>(animationsMutex_);
+    if (type == "entering") {
+      config = enteringAnimations_[tag];
+    } else if (type == "exiting") {
+      config = exitingAnimations_[tag];
+    } else if (type == "layout") {
+      config = layoutAnimations_[tag];
+    }
+    viewSharedValue = viewSharedValues_[tag];
+  }
+
+  jsi::Value layoutAnimationRepositoryAsValue =
+      rt.global()
+          .getPropertyAsObject(rt, "global")
+          .getProperty(rt, "LayoutAnimationRepository");
+  if (layoutAnimationRepositoryAsValue.isUndefined()) {
+    auto errorHandler = weakErrorHandler_.lock();
+    if (errorHandler != nullptr) {
+      errorHandler->setError(
+          "startLayoutAnimation called before initializing LayoutAnimationRepository");
+      errorHandler->raise();
+    }
+    return;
+  }
+  jsi::Function startAnimationForTag =
+      layoutAnimationRepositoryAsValue.getObject(rt).getPropertyAsFunction(
+          rt, "startAnimationForTag");
+  startAnimationForTag.call(
+      rt,
+      jsi::Value(tag),
+      jsi::String::createFromAscii(rt, type),
```

Comment:
This should make no difference because the only strings passed here should be `"entering"`, `"layout"`, or `"exiting"`, but maybe let's use UTF-8 everywhere just to be on the safe side. Nice catch!

Reply:
ASCII is valid UTF8, so why not 😄 
Types are always ASCII-only anyway, so it shouldn't really matter here, and it won't be a problem once we move to enums here either.
I'd leave it as is.